bundler/lib/bundler/dsl.rb

```diff
@@ -23,6 +23,7 @@ def self.evaluate(gemfile, lockfile, unlock)
 
     def initialize
       @source               = nil
+      @global_source_defined = false
```

Comment:
Using `@global_source_defined` as a flag to check whether a global source has been defined or not. any suggestion for a better name?

The `@sources` also has a `global_rubygems_source` and `global_path_source` which could be used instead of `@global_source_defined` to check if a global source has been added or not but I felt this would make it more complicated because `global_rubygems_source` is memoized with a new instance of `Bundler::Source::Rubygems`. Fixing this issue will require more work and I wasn't just sure about it. Let me know what you think 🤔

https://github.com/intuxicated/rubygems/blob/592dbf31a3344b24470f1c0f07a4078036f7cb76/bundler/lib/bundler/source_list.rb#L11-L13

Reply:
@intuxicated This approach seems fine too, but I think we can use a similar approach to the deprecation that warns about multiple global sources. For consistency and also because I think it will make it easier to figure out all the code that can be cleaned up once the deprecation is turned into an error.

I would define a `no_remotes?` method in `Bundler::Source::Rubygems` similar to the `multiple_remotes?` method. Like

```ruby
def no_remotes?
  @remotes.size == 0
end
```

Then I would define an `implicit_global_source?` method that checks whether the global source is defined implicitly with the above fallback (`rubygems_aggregate_class.new("allow_local" => true)`) and thus has no remotes. So, similarly to the `aggregate_global_source?` method, this one would be defined as

```ruby
def implicit_global_source?
  global_rubygems_source.no_remotes?
end
```

Finally I would add a deprecation to the `check_rubygems_source_safety` that uses the `implicit_global_source?` method in `Bundler::SourceList` with the same style as the one for multiple global sources, but that also warns this case.

Let me know if this approach makes sense.

Reply:
@deivid-rodriguez Thanks for the feedback. I'll refactor it.